Site Settings: Use feature checks rather than plan checks

obenland opened the pull request:
#### Changes proposed in this Pull Request

* Replaced uses of `siteHasScanProductPurchase` with feature checks.
* Removes `siteHasScanProductPurchase` selector.

#### Testing instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Navigate to /site-settings/jetpack
* Make sure you see the connection box on Atomic sites and Jetpack sites with scan feature <img width="1510" alt="image" src="https://user-images.githubusercontent.com/1398304/168914141-3a0864c8-1afd-41c9-aaa0-177c7426da7b.png">
* The changes in Navigation are only visible to sites on Jetpack below v9.9.

<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to p4TIVU-a66-p2
Depends on https://github.com/Automattic/wp-calypso/pull/63735

obenland committed:
Update snapshot

obenland committed:
Settings: Use feature check in navigation

obenland committed:
Settings: Use feature check in jetpack settings

obenland committed:
Settings: Use feature checks in security settings

obenland committed:
Remove unused selector

obenland committed:
Update snapshot

obenland committed:
Update prop name in test

gavande1 reviewed (approved):
This works as expected. 👍

obenland committed:
Update Jetpack Cloud menu items

Introduced in #63643

obenland commented on client/components/jetpack/sidebar/menu-items/jetpack-cloud.jsx:
```diff
@@ -20,7 +21,7 @@ export default ( { path } ) => {
 	const siteSlug = useSelector( getSelectedSiteSlug );
-	const hasScanProduct = useSelector( ( state ) => siteHasScanProductPurchase( state, siteId ) );
+	const hasScan = useSelector( ( state ) => siteHasFeature( state, siteId, WPCOM_FEATURES_SCAN ) );
 	const hasActiveRewind = useSelector( ( state ) => isRewindActive( state, siteId ) );
```
@bluefuton Checking the rewind status removes access to the settings menu for multisite jetpack sites that have backup purchases. Should this be a feature check for `backups` (2c728-pb/#plain) instead?

bluefuton replied:
Thanks @obenland, yes, that sounds sensible 👍 

I based my logic on `client/my-sites/site-settings/settings-security/main.jsx` that you're also updating in this PR.

bluefuton replied:
cc @Automattic/yamato-backup

obenland replied:
> I based my logic on client/my-sites/site-settings/settings-security/main.jsx that you're also updating in this PR.

Should I replace `ActiveRewind` with a feature check for `backups` there as well?

bluefuton replied:
It'd be good to have the Backup team's input on this - can you help @seear?

seear replied:
> Should I replace ActiveRewind with a feature check for backups there as well?

Yes, the intention here would have been to check for backup.

obenland committed:
Use backups feature check instead of active rewind

Captures sites that have the backups feature but that Rewind doesn't cover, like multisites.

DustyReagan reviewed (approved):
Tests well! 👍
